src/transformStateWithClones.js

```
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        if (
+          action.extraData &&
+          typeof action.extraData === 'object' &&
+          !Array.isArray(action.extraData)
+        ) {
+          currentState = { ...currentState, ...action.extraData };
+        }
+        break;
+
+      case 'removeProperties':
+        if (Array.isArray(action.keysToRemove)) {
+          currentState = { ...currentState };
+
+          for (const key of action.keysToRemove) {
+            delete currentState[key];
+          }
+        }
+        break;
+
+      default:
+        currentState = { ...currentState };
```

The `default` case does not perform error handling; it only clones the state. This violates the checklist requirement for the switch default to handle errors. Exact checklist item violated:

3. [CODE STYLE]: switch/case should always have default case for error handling.

Recommendation: Replace the current default branch with explicit error handling so unknown action types fail loudly, for example:
```
default:
  throw new Error(`Unknown action type: ${action.type}`);
```
This makes behavior predictable and testable.